### PR TITLE
Overhaul introductory section of docsite

### PR DESCRIPTION
**Pull request by lihaoyi**
Flesh out the intro to include answers to all the frequently asked questions, as determined by the recent discussion on reddit r/java and r/programming

Generally tried to make it more information dense and structured, covering the main points and elaborations in bullets, and with more detailed sections comparing against Maven, Gradle, and SBT. I tried to limit the descriptions to things that would be of most direct interest to Gradle/Maven users with zero knowledge of Mill or Scala, which means more empirical stuff (e.g. performance, IDE support, ease of customization) v.s. philosophical stuff (e.g. intuitive abstractions, pure FP, etc.)

I renamed all the `Scala_` and `Java_` pages to follow a consistent naming convention, and added forwarder aliases as necessary

### CODE REVIEW

File: docs/modules/ROOT/pages/Java_Intro_to_Mill.adoc [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  line of configuration in a Mill build is meaningful. e.g. adding a dependency is 1 line in
+  Mill, like it is in Gradle, and unlike the 5 line `<dependency>` declaration you find in Maven
+
+* **Mill helps keep your build understandable**: Unlike Gradle config statements that can
+  work in mysterious ways, Mill build configuration is done via vanilla methods, objects,
+  inheritance, and overrides. This means that impact of a Mill config statement is localized,
+  without "_spooky action at a distance_", and the exact meaning of any config statement is
+  a single "_jump-to-definition_" away in your IDE.
+
+
```
Comment (lihaoyi) [inline docs/modules/ROOT/pages/Java_Intro_to_Mill.adoc:65]: @lefou @lolgab do you guys use Gradle? I'm trying to come up with a third bullet point to put here, but not sure what it should be. What are the things a gradle user would find most interesting about Mill?

File: docs/modules/ROOT/partials/Intro_to_Mill_Header.adoc [hunk 2]
```diff
@@ -1,26 +1,26 @@
+{mill-github-url}[Mill] is a {language} build tool that aims for performance,
```
Comment (lefou) [inline docs/modules/ROOT/partials/Intro_to_Mill_Header.adoc:1]: I don't think we should claim it a "Java build tool" in one place and a "Scala build tool" in the other. I don't have a better wording yet, but we should aim for one.

Maybe just "build tool" or "generic build tool". It's running on the JVM, so naturally, it's "(primarily) targetting JVM languages".

File: docs/modules/ROOT/partials/Intro_to_Mill_Header.adoc [hunk 3]
```diff
@@ -1,26 +1,26 @@
...
 
-* **Mill builds are easy to extend by non-experts**: anyone can extend a Mill build
-with a few lines of code, without needing to learn a complicated plugin framework,
-allowing end users the freedom to fit their build logic to their own unique requirements
+* *Maintainability*: Mill build targets are self contained without side-effects, making it easy
+  to trace the data-flow of your build. Mill also provides better IDE support
+  (via xref:{language}_Installation_IDE_Support.adoc[IntelliJ or VSCode]) than any other build tool,
+  letting you use "_jump-to-definition_" to navigate around your build
+  as easily as any application codebase. This makes maintaining Mill builds
+  a far easier than the "_guess and grep_" workflows common with other build tools.
```
Comment (lefou) [inline docs/modules/ROOT/partials/Intro_to_Mill_Header.adoc:14]: The "better" might be arguable. Maybe, we just say "good" .

The "jump to definition" is most likely a unique selling point though.

File: docs/modules/ROOT/pages/Scala_Intro_to_Mill.adoc [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+  list of ways Mill improves upon SBT's performance is long, and at the command line you
+  can really feel it
+
+* **Mill builds are much easier to understand**: Your Mill build is made of bog-standard
+  ``object``s and ``def``s, rather than SBT's
+  https://eed3si9n.com/4th-dimension-with-sbt-013/[four-dimensional task matrix]. Your IDE's
+  "*jump-to-definition*" in Mill actually brings you to the implementation of a task, rather
+  than an SBT `taskKey` declaration. Customizing things is as simple as writing or overriding
+  `def`s. The net effect is that despite both tools' build files being written in Scala,
+  Mill's build files are much more
```
Comment (lefou) [inline docs/modules/ROOT/pages/Scala_Intro_to_Mill.adoc:52]: > ...are much more

Is there something missing at the end?

**Pull request merged**